src/guest_memory.rs

```diff
@@ -836,16 +841,20 @@ impl<T: GuestMemory> Bytes<GuestAddress> for T {
             // Check if something bad happened before doing unsafe things.
             assert!(offset <= count);
             if let Some(dst) = unsafe { region.as_mut_slice() } {
-                // This is safe cause `start` and `len` are within the `region`.
+                // This is safe cause `start` and `len` are within the `region`, and we manually
+                // record the dirty status of the written range below.
                 let start = caddr.raw_value() as usize;
                 let end = start + len;
-                loop {
+                let result = loop {
                     match src.read(&mut dst[start..end]) {
                         Ok(n) => break Ok(n),
                         Err(ref e) if e.kind() == std::io::ErrorKind::Interrupted => continue,
                         Err(e) => break Err(Error::IOError(e)),
                     }
-                }
+                };
+
+                region.bitmap().mark_dirty(start, len);
```

Comment:
There's a `region.write()` below that could benefit from a comment about why we don't manually update the bitmap there as well.

Reply:
Should we mention somewhere that the `GuestMemory` trait assumes the Bytes implementation of the regions will update the bitmap?